examples/boids.jl

```diff
@@ -0,0 +1,96 @@
+#----------------------------------------------------------------------
+# Boids algorithm visualization
+# Based on pseudocode at http://www.kfish.org/boids/pseudocode.html
+# Simulate flocking birds using simple rules.
+# Initial version developed and contributed by Iain Dunning (@IainNZ)
+# as part of the JuliaCon 2015 workshop
+# TODO:
+# - Add interactivity for parameters that govern behavior.
+# - Add a force that attracts to mouse pointer.
+#----------------------------------------------------------------------
+
+using Compose
+
+#----------------------------------------------------------------------
+
+immutable Boid
+    pos::Vector{Float64}
+    vel::Vector{Float64}
+end
+# Create new boid at random location and velocity
+Boid() = Boid(rand(2),rand(2)/1000)
+function draw(b::Boid)
+    (context(),
+        (context(),circle(b.pos[1]+b.vel[1],b.pos[2]+b.vel[2],0.005),fill("red")),
+        (context(),circle(b.pos[1],b.pos[2],0.01),fill("blue"))
+    )
+end
+
+#----------------------------------------------------------------------
+
+function main(window)
+    # Run at 30 FPS
+    eventloop = every(1/30)
```

Comment:
This should be `eventloop = fpswhen(window.alive, 30)`  though so that the 
animation stops when you close the window.